File: Haskell/beginner/1984.hs
-- https://judge.beecrowd.com/pt/problems/view/1984
-- O Enigma da Pronalândia

-- Imports
import Data.Char (isSpace) -- Não esqueça de importar isso!

-- Função auxiliar para remover espaços do início e do fim da string
trim :: String -> String
trim = f . f
  where f = reverse . dropWhile isSpace . reverse . dropWhile isSpace
-- Nota de Washu: A função 'f . f' significa aplicar 'f' duas vezes.
-- 'f' é 'reverter, tirar espaços do início'. Aplicar duas vezes
-- garante que os espaços sejam removidos do início e do fim da string original.

main :: IO ()
main = do
  pronMessage <- getLine
  -- Inverte a mensagem, depois "trimma" (remove espaços das bordas)
  -- e adiciona explicitamente a quebra de linha.
  putStr ((trim . reverse) pronMessage ++ "\n")

-- By: Fernando Serra
-- https://github.com/fernandosserra

File: C/beginner/1984.c
// https://judge.beecrowd.com/pt/problems/view/1984
// O Enigma da Pronalândia

// Libraries
#include <stdio.h>
#include <string.h>


// Main
int main(){

    char pronMessage[100];

    scanf("%s", pronMessage);

    for (int i = strlen(pronMessage) - 1; i >= 0; i--){
        printf("%c", pronMessage[i]);
    }
    printf("\n");

    return 0;
}

// By: Fernando Serra
// https://github.com/fernandosserra